File: week3/atakantekatan/21. Merge Two Sorted Lists.py
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(0)
        pointer = dummy

        while list1 and list2:
            if list1.val < list2.val:
                pointer.next = ListNode(list1.val)
                list1 = list1.next
                pointer = pointer.next
            elif list2.val < list1.val:
                pointer.next = ListNode(list2.val)
                list2 = list2.next
                pointer = pointer.next
            elif list1.val == list2.val:
                pointer.next = ListNode(list1.val)
                pointer.next.next = ListNode(list2.val)
                list1 = list1.next
                list2 = list2.next
                pointer = pointer.next.next
        
        if list1:
            pointer.next = list1
        elif list2:
            pointer.next = list2
        
        return dummy.next

File: week3/Can-Yavuz/21-merge-two-sorted-lists.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode MergeTwoLists(ListNode list1, ListNode list2) {
        ListNode dummyHead = new ListNode();
        ListNode current = dummyHead; 

        while (list1 != null && list2 != null) {
            if (list1.val < list2.val) {
                current.next = list1;
                list1 = list1.next; 
            } else {
                current.next = list2;
                list2 = list2.next; 
            }
            current = current.next; 
        }
        
        if (list1 != null) {
            current.next = list1;
        } else {
            current.next = list2;
        }
        
        return dummyHead.next;
    }
}